internal/configs/configmaps_test.go

```diff
@@ -285,6 +285,168 @@ func TestParseConfigMapAccessLogDefault(t *testing.T) {
 	}
 }
 
+func TestParseMGMTConfigMapError(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		configMap *v1.ConfigMap
+		msg       string
+	}{
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "",
+				},
+			},
+			msg: "Must have license-token-secret-name",
+		},
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{},
+			},
+			msg: "Must have license-token-secret-name key",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			_, _, err := ParseMGMTConfigMap(context.Background(), test.configMap, makeEventLogger())
+
+			if err == nil {
+				t.Errorf("Expected error, got nil")
+			}
+		})
+	}
+}
+
+func TestParseMGMTConfigMapWarnings(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		configMap *v1.ConfigMap
+		msg       string
+	}{
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "7",
+				},
+			},
+			msg: "enforce-initial-report is invalid",
+		},
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "",
+				},
+			},
+			msg: "enforce-initial-report set empty",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			_, configWarnings, _ := ParseMGMTConfigMap(context.Background(), test.configMap, makeEventLogger())
```

Comment:
is err checking silenced here?

Reply:
We are only testing warnings here, we have other test for errors

Reply:
license secret returns a both a warning and error but i did add this check, thanks.